unlock-js/src/walletService.js

```diff
@@ -289,20 +292,23 @@ export default class WalletService extends UnlockService {
    * @param {*} callback
    */
   async signData(account, data, callback) {
-    const isMetaMask = this.web3Provider && this.web3Provider.isMetaMask
-    const method = isMetaMask ? 'eth_signTypedData_v3' : 'eth_signTypedData'
-    // see https://github.com/MetaMask/metamask-extension/blob/c4caba131776ff7397d3a4071d7cc84907ac9a43/app/scripts/metamask-controller.js#L997
-    const sendData = isMetaMask ? JSON.stringify(data) : data
-
     try {
-      const result = await this.provider.send(method, [account, sendData])
-
-      callback(null, Buffer.from(result).toString('base64'))
+      const result = await this.unformattedSignTypedData(account, data)
+      return callback(null, Buffer.from(result).toString('base64'))
     } catch (err) {
       return callback(err, null)
     }
   }
 
+  async unformattedSignTypedData(account, data) {
```

Comment:
This is where the real changes begin. The `signData` function base64s its return value (I think this was originally used for tickets or something). I split it in 2:

- `unformattedSignTypedData`, which just does the signature so we can use it more generically
- `signData`, which uses the former function but returns it in base64 to maintain the expected interface.